http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -265,6 +265,22 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       )
     } yield ()
 
+  override def deleteAllRuntimes(userInfo: UserInfo, workspaceId: WorkspaceId, deleteDisk: Boolean)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      runtimes <- RuntimeServiceDbQueries
+        .listRuntimesForWorkspace(Map.empty, true, None, Some(workspaceId), None)
+        .map(_.toList)
+        .transaction
+      // In ListRuntimeResponse2, the runtimeName is weirdly named clusterName instead
+      runtime_names = runtimes.map(r => r.clusterName)
+
+      _ <- runtime_names.traverse(runtime_name => deleteRuntime(userInfo, runtime_name, workspaceId, deleteDisk))
```

Comment:
I don't think this is exactly what we want. This gets the job done, but it causes a lot of extra DB queries, which can be expensive. See what I did for apps, I split the default `delete` function into two parts, the db lookup, then a function that takes the app for the rest of the flow. Should be possible here to make a third function that takes a `runtime`

Reply:
Good point, thanks for the catch, I will look closer at your PR and update :)

Reply:
Actually, the db queries are different between the get active runtime, and the list runtime. One returns a `ClusterRecord` that contains a lot more information than the `ListRuntimeResponse2`, so separating the two seems a bit delicate

Reply:
I scratched my head for 10 more minutes and I do not see an obvious way of not querying for the active runtime record every time 🤔